src/projections.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef SPHERELY_PROJECTIONS_H_
+#define SPHERELY_PROJECTIONS_H_
+
+#include <s2/s2latlng.h>
+#include <s2/s2projections.h>
+#include <s2geography.h>
+
+#include "pybind11.hpp"
+
+namespace py = pybind11;
+namespace s2geog = s2geography;
+using namespace spherely;
+
+class Projection {
+public:
+    Projection(std::shared_ptr<S2::Projection> projection)
+        : m_s2_projection(std::move(projection)) {}
+
+    std::shared_ptr<S2::Projection> s2_projection() {
+        return m_s2_projection;
+    }
+
+    static Projection lnglat() {
+        return Projection(std::move(s2geog::lnglat()));
+    }
+    static Projection pseudo_mercator() {
+        return Projection(std::move(s2geog::pseudo_mercator()));
+    }
+    static Projection orthographic(double longitude, double latitude) {
+        return Projection(
+            std::move(s2geog::orthographic(S2LatLng::FromDegrees(latitude, longitude))));
+    }
+
+private:
+    std::shared_ptr<S2::Projection> m_s2_projection;
```

Comment:
Not sure why a shared pointer is needed (apart from that this is what is used in s2geography), but that's not a big deal anyway. Feel free to ignore this comment.

Reply:
We indeed need a shared pointer to pass to s2geography (and it is also what is returned by the s2geography constructors), so then I though that might be easier to just store the shared_ptr on the object, instead of unpacking it / packing it again in a shared_ptr ?